pkg/database/vercode.go

```diff
@@ -97,11 +97,50 @@ func (v *VerificationCode) AfterCreate(scope *gorm.Scope) {
 				SET codes_issued = authorized_app_stats.codes_issued + 1
 		`
 
-		day := time.Now().UTC().Truncate(24 * time.Hour)
-		if err := scope.DB().Exec(sql, day, v.IssuingAppID).Error; err != nil {
+		if err := scope.DB().Exec(sql, date, v.IssuingAppID).Error; err != nil {
 			scope.Log(fmt.Sprintf("failed to update stats: %v", err))
 		}
 	}
+
+	// Update the per-realm stats.
+	sql := `
+		INSERT INTO realm_stats(date, realm_id, codes_issued)
+			VALUES ($1, $2, 1)
+		ON CONFLICT (date, realm_id) DO UPDATE
+			SET codes_issued = realm_stats.codes_issued + 1
+	`
+
+	if err := scope.DB().Exec(sql, date, v.RealmID).Error; err != nil {
+		scope.Log(fmt.Sprintf("failed to update stats: %v", err))
+	}
+}
+
+// claim checks the VerificationCode, marks as claimed, and updates the
+// per-realm stats. If any of that fails, an error code is returned.
+func (v *VerificationCode) claim(db *Database, tx *gorm.DB, verCode string, realmID uint) error {
```

Comment:
This API is a little weird to me now, and I'm not sure the comment matches. While this sets `Claimed: true` on the Verification Code, it does not _save_ that code in the database. The caller still needs to call `db.Save(v)`. That feels subtle and could lead to some bugs. Furthermore, we're updating the realm stats _before_ successfully marking the code as claimed. I'm thinking this flow should be more like:

1. Expiration checks
1. Claimed check
1. Mark as claimed on struct
1. Save struct to database (stop if error)
1. Update stats
1. Return

Alternatively, we need to document this better. What do you think?